grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandler.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.grpc;
+
+import com.google.common.collect.ImmutableMap;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.grpc.protocol.GrpcHeaderNames;
+import com.linecorp.armeria.internal.common.grpc.GrpcStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.grpc.Status;
+import io.grpc.Status.Code;
+
+/**
+ * Error handler which maps grpc response to {@link HttpStatus}.
+ */
+@FunctionalInterface
+@UnstableApi
+public interface UnframedGrpcErrorHandler {
+
+    /**
+     * Returns a plain text or json response based on the content type.
+     */
+    static UnframedGrpcErrorHandler of() {
+        return (ctx, status, response) -> {
+            final MediaType grpcMediaType = response.contentType();
```

Comment:
should we return json response in that case?

Reply:
Well, I'm not sure about this as well. It's probably okay to send the plain response because we were sending it. /cc @ikhoon

Reply:
I think gRPC clients will not try to parse if a status is non-OK.
So either way is fine to me. But I prefer JSON response because it is easier to parse if some clients need it.